sympy/polys/matrices/domainmatrix.py

```diff
@@ -36,6 +36,26 @@ def from_rep(cls, ddm):
 
     @classmethod
     def from_list_sympy(cls, nrows, ncols, rows, **kwargs):
+        """Convert a list of lists of Expr into a DomainMatrix
+
+        Examples
+        ========
+        Define a Domain Matrix on the field of Integers
+
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> A = DomainMatrix.from_list_sympy(1, 2, [[1, 0]])
+        >>> A
+        DomainMatrix([[1, 0]], (1, 2), ZZ)
+
+        We now define a Domain Matrix on the field of Rational Numbers
```

Comment:
The first example is sufficient. Another example is not needed.

It would be better to explain that this uses `construct_domain` (and have a see also for `construct_domain`) to choose a domain and converts all Expr to that domain.